### ISSUE TITLE
Upgrade node-forge to address vulnerabilities

### ISSUE THREAD

**Issue by damieng** [getyoti/yoti-node-sdk#327]
Hi.

It seem this package is using a very old version of node-forge that has 2 high vulnerabilities in it. This is blocking us from being able to ship with the yoti dependency.

The full report is

```
npm audit

                       === npm audit security report ===


                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance


  Moderate        Open Redirect in node-forge

  Package         node-forge

  Patched in      >=1.0.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-8fr3-hfg3-gpgp


  Low             Prototype Pollution in node-forge debug API.

  Package         node-forge

  Patched in      >=1.0.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-5rrq-pxf6-6jx5


  Moderate        Improper Verification of Cryptographic Signature in
                  `node-forge`

  Package         node-forge

  Patched in      >=1.3.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-2r2c-g63r-vccr


  High            Improper Verification of Cryptographic Signature in
                  node-forge

  Package         node-forge

  Patched in      >=1.3.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-x4jg-mjrx-434g


  High            Improper Verification of Cryptographic Signature in
                  node-forge

  Package         node-forge

  Patched in      >=1.3.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-cfm4-qjh2-4765


  Low             URL parsing in node-forge could lead to undesired behavior.

  Package         node-forge

  Patched in      >=1.0.0

  Dependency of   yoti

  Path            yoti > node-forge

  More info       https://github.com/advisories/GHSA-gf8q-jrpm-jvxq

found 6 vulnerabilities (2 low, 2 moderate, 2 high) in 66 scanned packages
  6 vulnerabilities require manual review. See the full report for details.
```

**Comment by laurent-yoti** [comment]
Hi @damieng

Thanks for raising the issue with us, and apologies for the delay regarding any response. 
We have been aware of the issue and willing to address it for some time, yet it has never been prioritised accordingly.

We are currently in the process of updating dependencies, which will be including in the next version **v3.23.0**.

You can follow update in #329, and we are targeting a release next week.
Thanks

**Comment by laurent-yoti** [comment]
Hi @damieng,

It took the team a little longer than anticipated as we actually released a major version **v4.0.0** and could eliminate more dependencies.
As a result there is a few breaking changes, please refer to the [v4.0.0 release notes](https://github.com/getyoti/yoti-node-sdk/releases/tag/v4.0.0) to find out.

Thanks

**Comment by damieng** [comment]
Cool, thanks, will switch to new version.
